trpcultivate_phenotypes/src/Form/PhenodataBackupForm.php

```diff
@@ -0,0 +1,327 @@
+   */
+  protected $tripal_importer;
+
+  /**
+   * Users.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected AccountInterface $user;
+
+  /**
+   * Form mode.
+   *
+   * @var string
+   */
+  private string $form_mode;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity Type manager service.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Configuration factory service.
+   * @param \Drupal\tripal\TripalImporter\PluginManagers\TripalImporterManager $tripal_importer_manager
+   *   Tripal importer plugin manager.
+   * @param \Drupal\Core\Session\AccountInterface $user
+   *   Drupal users.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    ConfigFactoryInterface $config_factory,
+    TripalImporterManager $tripal_importer_manager,
+    AccountInterface $user,
+  ) {
+
+    $this->service_EntityTypeManager = $entity_type_manager;
+    $this->service_ConfigFactory = $config_factory;
+    $this->tripal_importer = $tripal_importer_manager;
+    $this->user = $user;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('config.factory'),
+      $container->get('tripal.importer'),
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+
+    $form = parent::form($form, $form_state);
+
+    $entity = $this->getEntity();
+
+    // Entity configuration ID number.
+    $form['id'] = [
+      '#type' => 'hidden',
+      '#default_value' => $entity->id(),
+      '#value' => uniqid(),
+    ];
+
+    if ($entity->isNew()) {
+      // Synchronize the file types defined by the importer that will be the
+      // main source of future backup data collection file.
+      $importer_plugin_definition = $this->tripal_importer
+        ->getDefinition('trpcultivate-phenotypes-share-importer');
+      $importer_file_extension = implode(' ', $importer_plugin_definition['file_types']);
+
+      $config_backup_dir = $this->service_ConfigFactory
+        ->get('trpcultivate_phenotypes.settings')
+        ->get('trpcultivate.phenotypes.directory.data_backup');
+      $backup_dir = $config_backup_dir ?? 'public:://phenotype-backups/';
+
+      $form['backup_file'] = [
+        '#type' => 'managed_file',
+        '#title' => 'Data File',
+        '#description' => $this->t('Select data file to backup. Only [@ext] file extensions are allowed.', ['@ext' => $importer_file_extension]),
+        '#upload_location' => $backup_dir,
+        '#multiple' => FALSE,
+        '#required' => TRUE,
+        '#upload_validators' => [
+          'FileExtension' => [
+            'extensions' => $importer_file_extension,
+          ],
+        ],
+      ];
```

Comment:
This should be reused rather then having the if isNew() and essentially two forms. The following will allow this form to show the file uploaded on edit but for it to be disabled (cannot remove or change).

```suggestion
      $fids = $entity->get('file_id');
      $fids = (is_string($fids)) ? [$fids] : $fids;
      $form['backup_file'] = [
        '#type' => 'managed_file',
        '#title' => 'Data File',
        '#description' => $this->t('Select data file to backup. Only [@ext] file extensions are allowed.', ['@ext' => $importer_file_extension]),
        '#upload_location' => $backup_dir,
        '#multiple' => FALSE,
        '#required' => TRUE,
        '#upload_validators' => [
          'FileExtension' => [
            'extensions' => $importer_file_extension,
          ],
        ],
        '#value' => [
          'fids' => $fids,
        ],
        '#disabled' => (empty($fids)) ? FALSE : TRUE,
      ];
```

Reply:
Reused the field, this is awesome.. I modified the value key a bit since using this setup the file field seems to upload the file into the void